### PR TITLE
Drivetrain

### PR DESCRIPTION
**Pull request by AlexaZoubrilov**
Revised Drivetrain according to comments by Mr. Bross

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    gearShift = new Solenoid(DriveConstants.kDriveSolenoid);
+
+    addChild("Left Motors", m_leftMotors);
+    addChild("Right Motors", m_rightMotors);
+    addChild("Drivetrain", m_Drive);
+    addChild("Shift Gears", gearShift);
+
+  }
+
+private void encoderInit(CANEncoder encoder) {
```
Comment (rjbohinski) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: I would suggest fixing the indentation in this file.

In VSCode you can do this by:
- Shortcut `Shift` + `Alt` + `F` on Windows **OR**
- `Ctrl` + `Shift` + `P` and type format

**Review by rjbohinski — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    resetRightEncoders();
+  }
+
+private void motorInit(CANSparkMax motor, boolean invert) {
+    motor.restoreFactoryDefaults(); // Resets settings in motor in case they are changed
+    motor.setIdleMode(IdleMode.kBrake); // Sets the motors to brake mode from the beginning
+    motor.setSmartCurrentLimit(kCurrentLimit);
+
+    encoderInit(motor.getEncoder()); // Initializes encoder within motor
+  }
```
Comment (zbross) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: This method should also be setting the motor to be inverted or not according to the parameter passed through to "invert"

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    return (m_leftMotorEncoder1.getPosition() + m_leftMotorEncoder2.getPosition()) / 2.0;
+  }
+
+// Gets right side distance by averaging right encoders
+private double getRightEncoderAverage() {
+    return (m_rightMotorEncoder1.getPosition() + m_rightMotorEncoder2.getPosition()) / 2.0;
+  }
+
+// Gets average distance by averaging the right and left distance
+public double getDistance() {
```
Comment (zbross) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: Let's rename this to getAverageDistance() to make it clear to other users how we're using it.

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    motorInit(m_leftMotor2, DriveConstants.kLeftEncoderInverted); 
+    m_leftMotors = new SpeedControllerGroup(m_leftMotor1, m_leftMotor2);
+    m_rightMotors = new SpeedControllerGroup(m_rightMotor1, m_rightMotor2);
+
+    m_Drive = new DifferentialDrive(m_leftMotors, m_rightMotors);
+
+     // adds options to the drive type
+     m_chooser.addOption("Tank Drive", "Tank");
+     m_chooser.addOption("Arcade Drive", "Arcade");
+     m_chooser.addOption("Trigger Drive", "Trigger");
```
Comment (zbross) [inline src/main/java/frc/robot/subsystems/Drivetrain.java]: After adding the options to the chooser, we also need to send it to the SmartDashboard by using
SmartDashboard.putData("Drive Control", m_chooser);
We can do this right away or later, but adding it after the various addChilds would make sense to me.

**Review by zbross — CHANGES_REQUESTED** [review]
Looking good!  We've got just a few small things to update and then you should be good to merge.

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
```
Comment (zbross) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:73]: After adding the options to the chooser, we also need to send it to the SmartDashboard.
```suggestion

SmartDashboard.putData("Drive Control", m_chooser);
```

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    // Resets all drive encoders
+    resetLeftEncoders();
+    resetRightEncoders();
+  }
+
+private void motorInit(CANSparkMax motor, boolean invert) {
+    motor.restoreFactoryDefaults(); // Resets settings in motor in case they are changed
+    motor.setIdleMode(IdleMode.kBrake); // Sets the motors to brake mode from the beginning
+    motor.setSmartCurrentLimit(kCurrentLimit);
+
```
Comment (zbross) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:122]: This method should also be setting the motor to be inverted or not according to the parameter passed through to "invert"
```suggestion

motor.setInverted(invert);
```

**Review by zbross — CHANGES_REQUESTED** [review]
Looking good!  Just a few small updates to get our core functionality up and running.

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 3 above)
Comment (ShrutiMandre) [reply src/main/java/frc/robot/subsystems/Drivetrain.java]: The method has been renamed, thanks.

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 5 above)
Comment (ShrutiMandre) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:73]: added it to smart dashboard

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 6 above)
Comment (ShrutiMandre) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:122]: added the invert parameter to the method

**Review by zbross — APPROVED** [review]

**Pull request merged**